contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -48,6 +48,32 @@ CREATE OR REPLACE AGGREGATE sys.tsql_select_for_xml_text_agg(
     FINALFUNC = tsql_query_to_xml_text_ffunc
 );
 
+CREATE OR REPLACE FUNCTION sys.tsql_select_for_xml_result(res XML)
+RETURNS setof XML AS
+$$
+BEGIN
+IF res IS NOT NULL THEN
+    return next res;
+ELSE
+    return;
+END IF;
+END;
+$$
+LANGUAGE plpgsql;
```

Comment:
Can it be IMMUTABLE or STABLE?

Reply:
I wasn't sure originally, but I think it should be okay to set it to IMMUTABLE